service-registry/src/main/java/org/apache/servicecomb/serviceregistry/task/ServiceCenterTask.java

```diff
@@ -62,7 +75,27 @@ public void onMicroserviceInstanceHeartbeatTask(MicroserviceInstanceHeartbeatTas
     if (task.getHeartbeatResult() != HeartbeatResult.SUCCESS) {
       LOGGER.info("read MicroserviceInstanceHeartbeatTask status is {}", task.taskStatus);
       onException();
+      if (!safeMode && consecutiveFailedTimes.incrementAndGet() > checkTimes) {
+        LOGGER.warn("service center is unavailable, enter safe mode");
+        eventBus.post(new SafeModeChangeEvent(true));
+      }
+      if (consecutiveSucceededTimes.get() != 0) {
```

Comment:
directly use consecutiveSucceededTimes.set(0) is simpler.

Reply:
done